extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/tracing/client/ClientTracingFilter.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.opentelemetry.tracing.client;
+
+import java.io.IOException;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+
+import org.jboss.resteasy.client.jaxrs.internal.ClientRequestContextImpl;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+
+@Priority(Priorities.HEADER_DECORATOR)
+public class ClientTracingFilter implements ClientRequestFilter, ClientResponseFilter {
+    private static final TextMapPropagator TEXT_MAP_PROPAGATOR = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
+
+    private Tracer tracer;
+    private Span clientSpan;
```

Comment:
Will this usage run into trouble? is it possible to have the same filter used across multiple requests?